nyx_client/nyx_client/circles.py

```diff
@@ -0,0 +1,113 @@
+# Copyright © 2024 IOTIC LABS LTD. info@iotics.com
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://github.com/Iotic-Labs/nyx-sdk/blob/main/LICENSE
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class RemoteHost:
+    """Represents a remote host on the network."""
+
+    def __init__(self, name: str, did: str):
+        """Builds a RemoteHost.
+
+        Args:
+            name: the name of the organization
+            did: the did of the remote host
+        """
+        self.name = name
+        self.did = did
+
+    def __repr__(self):
+        return f"RemoteHost({self.name}, {self.did})"
+
+
+class Circle:
```

Comment:
could also be a frozen dataclass?

Reply:
I think it could be useful for making changes to the circle as update_circle takes a `Circle` we probably want people to update this class?

Reply:
Ah right - because they can change the description too. :+1: